Lib/ipaddress.py

```diff
@@ -1834,8 +1854,23 @@ def __init__(self, address):
         addr_str = str(address)
         if '/' in addr_str:
             raise AddressValueError("Unexpected '/' in %r" % address)
+        addr_str, self._scope_id = self._split_scope_id(addr_str)
+
         self._ip = self._ip_int_from_string(addr_str)
 
+    def __str__(self):
+        ip_str = super().__str__()
+        return ip_str + '%' + self._scope_id if self._scope_id else ip_str
+
+    @property
+    def scope_id(self):
+        """Identifier of a particular zone of the address's scope.
```

Comment:
So from looking through [the RFC](https://tools.ietf.org/html/rfc4007), I think the standard terminology is: this identifier identifies a "zone", not a scope. (This description in the docstring looks exactly correct!) So a good name would be `zone_id`.

(In fact that's exactly the name the RFC uses when discussing the syntax.)

Reply:
I agree with you, [RFC 4007](https://tools.ietf.org/html/rfc4007#section-11) calls this part "zone_id".
At the same time, in existing Python code and documentation "scope id" is widely used too, e.g.:
[socket module](https://github.com/python/cpython/blob/master/Modules/socketmodule.c#L1879-L1881), [socket module](https://github.com/python/cpython/blob/master/Modules/socketmodule.c#L6649), [socket module tests](https://github.com/python/cpython/blob/master/Lib/test/test_socket.py#L1693-L1744), [socket module docs](https://docs.python.org/3/library/socket.html).
That examples include using of "scope_id" as part of AF_INET6 address family representation.
So, as for me, naming of attribute is the subject for discussion.